fe1-web/model/objects/Chirp.ts

```diff
@@ -0,0 +1,87 @@
+import { Timestamp } from './Timestamp';
+import { Hash } from './Hash';
+
+/**
+ * Object to represent a Chirp.
+ */
+
+export interface ChirpState {
+  id: string;
+  sender: string;
+  text: string;
+  time: number;
+  likes: number;
+  parentId: string;
+}
+
+export class Chirp {
+  public readonly id: Hash;
+
+  // The sender's public key
+  public readonly sender: string;
+
+  // The text of the chirp
+  public readonly text: string;
+
+  // The time where the chirp was posted
+  public readonly time: Timestamp;
+
+  // The number of likes
+  public readonly likes: number;
+
+  // The id of the parent chirp (if it is a reply)
+  public readonly parentId?: Hash;
+
+  constructor(obj: Partial<Chirp>) {
+    if (obj === undefined || obj === null) {
+      throw new Error('Error encountered while creating a Chirp object: '
+        + 'undefined/null parameters');
+    }
```

Comment:
Per your question, you can test this behavior by forcing a cast. 
E.g. `const obj = null as Partial<Chirp>`.

This is basically the TypeScript way of telling the language "shut up and trust me, I know what I'm doing"

Reply:
Thanks ! I didn't think about that.

Reply:
Yup, sorry, that's the kind of details I tend to forget when switching between languages 😅